portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotosContainerResource.java

```diff
@@ -73,4 +83,78 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(getAlbums(), getPhotos());
   }
+
+  public void transmogrifyAlbums(int maxSize){
+    transmogrifyAlbums(maxSize, true);
+  }
+
+  // Coerce the albums of the transfer using the specification provided, e.g.
+  // limiting max album size or grouping un-collected photos into a root album
+  public void transmogrifyAlbums(int maxSize, boolean allowRootPhotos){
+      ensureRootAlbum(allowRootPhotos);
+      ensureAlbumSize(maxSize);
+  }
+
+  public void ensureAlbumSize(int maxSize){
+    if (maxSize == -1){
+      // No max size; no need to go through that code.
+      return;
+    }
+    // Group photos by albumId
+    Multimap<String, PhotoModel> albumGroups = ArrayListMultimap.create();
+    for (PhotoModel photo: photos){
+      albumGroups.put(photo.getAlbumId(), photo);
+    }
+    // Go through groups, splitting up anything that's too big
+    for(Entry<String,Collection<PhotoModel>> entry : albumGroups.asMap().entrySet()){
+      if (entry.getValue().size() > maxSize) {
+        for(PhotoAlbum album: albums){
+          if (album.getId() != entry.getKey()){
+            break;
+          }
+          // Create new partial album objects and reassign photos to those albums
+          List<PhotoAlbum> newAlbums = album.split(-Math.floorDiv(- entry.getValue().size(), maxSize));
```

Comment:
I may be reading this wrong but if there's 10001 photos, does this create 2 new albums? Where does the 10001th photo get put?

Reply:
the `-floordiv` thing is equivalent to ceilDiv, which for some reason isn't in the java math library. If we have a max bucket size of 5 and have 10 things, we'll have two buckets, if we have 11, we'll have 3 (5+5+1)

Reply:
I always thought it would return the integer less than the quotient https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#floorDiv-int-int-

Reply:
right, we negate the numerator, which makes the result of the floor div round to the next most-negative number if it's not an even division, and then we negate that, which makes the resulting value just the ceiling